### PR TITLE
Add blog post about the `psp-host-namespaces` policy

### PR DESCRIPTION
**Pull request by ereslibre**
Closes https://github.com/kubewarden/kubewarden.io/issues/14

### CODE REVIEW

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  This example would allow host ports `80`, `443` and the range
+  `8000-9000` in the Pod `.spec.containers[].ports[].hostPort` attributes.
+
+By default all settings are disallowed, so if you want to allow the
+usage of, say, the host network and the host PID namespace usage, the
+policy would need to have the following settings:
+
+```json
+{ "allow_host_network": true, "allow_host_pid": true }
+```
```
Comment (flavio) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:76]: Write that as `yaml`, which is more readable

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+### Running the policy
+
+The policy validates Pods at creation time.
+
+Let's run a couple of tests with [our CLI tool,
+`kwctl`](https://github.com/kubewarden/kwctl/).
+
+You will need at least `kwctl` version 0.1.7 to execute this examples.
+
+```console
```
Comment (flavio) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:87]: Let's start by evaluating a Pod creation request which has `host PID` enabled:

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    kwctl run -r - registry://ghcr.io/kubewarden/policies/psp-host-namespaces:v0.0.7 | jq
+{
+  "uid": "1299d386-525b-4032-98ae-1949f69f9cfc",
+  "allowed": false,
+  "status": {
+    "message": "Pod has host PID enabled, but this is not allowed"
+  }
+}
+```
+
```
Comment (flavio) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:98]: As you can see, the request is rejected and a proper explanation message is returned to the user.

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+Settings can also be provided as an empty string or an empty object in
+this case:
```
Comment (flavio) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:100]: I would remove "in this case"

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+$ curl https://raw.githubusercontent.com/kubewarden/psp-host-namespaces/v0.0.7/test_data/pod_host_pid_enabled.json | \
+    kwctl run --settings-json '{}' -r - registry://ghcr.io/kubewarden/policies/psp-host-namespaces:v0.0.7 | jq
+{
+  "uid": "1299d386-525b-4032-98ae-1949f69f9cfc",
+  "allowed": false,
+  "status": {
+    "message": "Pod has host PID enabled, but this is not allowed"
+  }
+}
+```
```
Comment (flavio) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:124]: I would drop this case.

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+As you can see it rejected our pod. This is because settings are
+defaulted to the most restrictive ones. Now, let's enable the PID
```
Comment (flavio) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:127]: Maybe replace `restrictive` with security-conscious.

I would also create a new paragraph for "Now, let's enable..."

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+Stay tuned, because during the following weeks we will implement the
+[rest of the Pod Security Policies in separate, small Kubewarden
+policies](https://github.com/kubewarden/policy-hub/issues/32).
+
+You can use [this GitHub
+search](https://github.com/kubewarden/?q=psp&type=&language=&sort=) to
+find out what Kubewarden policies are implementing Pod Security
+Policies. You can also discover more policies in the [Policy
+Hub](https://hub.kubewarden.io/).
```
Comment (flavio) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:151]: I put the highlight only on policy hub.

**Review by flavio — CHANGES_REQUESTED** [review]
I like it, I just left some minor comments

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md (hunk 7 above)
Comment (ereslibre) [reply content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:151]: Do you mean dropping the GitHub search?

Comment (ereslibre) [reply content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md]: Moved it to a new paragraph, is that better? Or should I drop the GitHub search?

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
+---
+title: Introducing the PSP host namespaces policy
+authors:
+    - Rafael Fernández López
+date: 2021-06-11
+---
+
+As you probably know, Kubernetes Pod Security Policies [are being
```
Comment (kkaempf) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:8]:
```suggestion
As you probably know, Kubernetes Pod Security Policies (PSPs) [are being
```

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+deprecated in Kubernetes
+1.21](https://github.com/kubernetes/kubernetes/blob/a3abd06ad53b2f02dcb6e060c7606ceda41f44af/CHANGELOG/CHANGELOG-1.21.md#deprecation-of-podsecuritypolicy)
+-- although these APIs will be served until Kubernetes 1.25 it's a
+good time to start thinking about what you will use to replace them.
+
+At Kubewarden we have an ongoing effort to replace the Pod Security
+Policies with small, targeted Kubewarden policies.
+
+Up until now, we have implemented some policies that replace some Pod
```
Comment (kkaempf) [inline content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:17]: where are these PSPs that are being replaced ? Can you add a link here ?

**Review by kkaempf — CHANGES_REQUESTED** [review]

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md (hunk 9 above)
Comment (ereslibre) [reply content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md:17]: Yes, they are listed on the next lines :)

**Review by kkaempf — APPROVED** [review]
Ship it ! :wink:

**Review by flavio — APPROVED** [review]
shit it! :rocket:

File: content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md (hunk 7 above)
Comment (flavio) [reply content/blog/2021/06/introducing-the-psp-host-namespaces-policy.md]: it's perfect now

**Pull request merged**